modules/related-posts/jetpack-related-posts.php

```diff
@@ -769,7 +769,7 @@ public function get_for_post_id( $post_id, array $args ) {
 			$options['size'] = $args['size'];
 		}
 
-		if ( ! $options['enabled'] || 0 == (int)$post_id || empty( $options['size'] ) || get_post_status( $post_id) !== 'publish' ) {
```

Comment:
@aldavigdis Could you tell me more about removing the `enabled` check? This is causing a regression in that we have a filter so folks can selectively disable RPs.

The published check was added in https://github.com/Automattic/jetpack/pull/9495 as a way to fix #9494 for a 400 HTTP request in the Block Editor for a published post.

See https://github.com/Automattic/jetpack/pull/11560

See #11546